### DISCUSSION TITLE
Cannot update after v4.2.0

### DISCUSSION THREAD

**Discussion by n-aggelidis** [nextcloud/all-in-one#2070] [Questions] [answered]
Hello there!

I have the problem that my AIO interface shows that all containers are up to date, but the latest version is now 4.4.1 and i am worried that the update process might not function correctly. I am running on the latest arm channel with Caddy as the reverse proxy and the mastercontainer log shows this error here:

[Fri Feb 24 12:19:27.786015 2023] [php:notice] [pid 115] [client 127.0.0.1:42496] Could not get digest of container nextcloud/all-in-one:latest-arm64 Client error: `GET https://registry-1.docker.io/v2/nextcloud/all-in-one/manifests/latest-arm64` resulted in a `404 Not Found` response:\n{"errors":[{"code":"MANIFEST_UNKNOWN","message":"manifest unknown"}]}\n\n
[Fri Feb 24 12:19:28.042974 2023] [php:notice] [pid 115] [client 127.0.0.1:42496] Could not get digest of container nextcloud/aio-apache:latest-arm64 Client error: `GET https://registry-1.docker.io/v2/nextcloud/aio-apache/manifests/latest-arm64` resulted in a `404 Not Found` response:\n{"errors":[{"code":"MANIFEST_UNKNOWN","message":"manifest unknown"}]}\n\n
[Fri Feb 24 12:19:28.294599 2023] [php:notice] [pid 115] [client 127.0.0.1:42496] Could not get digest of container nextcloud/aio-collabora:latest-arm64 Client error: `GET https://registry-1.docker.io/v2/nextcloud/aio-collabora/manifests/latest-arm64` resulted in a `404 Not Found` response:\n{"errors":[{"code":"MANIFEST_UNKNOWN","message":"manifest unknown"}]}\n\n
[Fri Feb 24 12:19:28.551863 2023] [php:notice] [pid 115] [client 127.0.0.1:42496] Could not get digest of container nextcloud/aio-nextcloud:latest-arm64 Client error: `GET https://registry-1.docker.io/v2/nextcloud/aio-nextcloud/manifests/latest-arm64` resulted in a `404 Not Found` response:\n{"errors":[{"code":"MANIFEST_UNKNOWN","message":"manifest unknown"}]}\n\n
[Fri Feb 24 12:19:28.809754 2023] [php:notice] [pid 115] [client 127.0.0.1:42496] Could not get digest of container nextcloud/aio-postgresql:latest-arm64 Client error: `GET https://registry-1.docker.io/v2/nextcloud/aio-postgresql/manifests/latest-arm64` resulted in a `404 Not Found` response:\n{"errors":[{"code":"MANIFEST_UNKNOWN","message":"manifest unknown"}]}\n\n
[Fri Feb 24 12:19:29.062956 2023] [php:notice] [pid 115] [client 127.0.0.1:42496] Could not get digest of container nextcloud/aio-redis:latest-arm64 Client error: `GET https://registry-1.docker.io/v2/nextcloud/aio-redis/manifests/latest-arm64` resulted in a `404 Not Found` response:\n{"errors":[{"code":"MANIFEST_UNKNOWN","message":"manifest unknown"}]}\n\n
[Fri Feb 24 12:19:29.314552 2023] [php:notice] [pid 115] [client 127.0.0.1:42496] Could not get digest of container nextcloud/aio-fulltextsearch:latest-arm64 Client error: `GET https://registry-1.docker.io/v2/nextcloud/aio-fulltextsearch/manifests/latest-arm64` resulted in a `404 Not Found` response:\n{"errors":[{"code":"MANIFEST_UNKNOWN","message":"manifest unknown"}]}\n\n
[Fri Feb 24 12:19:29.601455 2023] [php:notice] [pid 115] [client 127.0.0.1:42496] Could not get digest of container nextcloud/aio-imaginary:latest-arm64 Client error: `GET https://registry-1.docker.io/v2/nextcloud/aio-imaginary/manifests/latest-arm64` resulted in a `404 Not Found` response:\n{"errors":[{"code":"MANIFEST_UNKNOWN","message":"manifest unknown"}]}\n\n

Thank you very much for your help!

**Comment by Zoey2936** [comment, marked as answer]
can you try using the latest instead of the lastest-arm64 image tag please?

**Reply by n-aggelidis** [reply]
I thought latest does not work with arm architecture? But now i see that in the Readme the latest-arm64 is not mentioned anymore, so has that changed?

**Reply by Zoey2936** [reply]
We introduced multiarch images, this is what breaked your update check

**Reply by n-aggelidis** [reply]
Ah i see, but for existing arm deploys it was supposed to still be working, but here the error occurs? So i can just rerun the Aio mastercontainer with the latest image instead and the cloud should continue working normally?

**Reply by Zoey2936** [reply]
on arm64 instances latest and latest-arm64 should work (they are the same), but it seems that latest-arm64 is not working, I try to fix it, so yes latest-arm64 was supposed to continue working, with switching to the now preferred latest tag everything should work

**Reply by n-aggelidis** [reply]
Okay, i will test it, thank you very much for your help!

**Reply by n-aggelidis** [reply]
Yes it worked, thank you!

**Comment by Zoey2936** [comment]
@szaimen `curl -H "Authorization: Bearer $(curl "https://auth.docker.io/token?service=registry.docker.io&scope=repository:nextcloud/all-in-one:pull" -s | jq -r .token)" -H "Accept: application/vnd.docker.distribution.manifest.v2+json" https://registry-1.docker.io/v2/nextcloud/all-in-one/manifests/latest-arm64 -Is | grep docker-content-digest` (the old Accept header) seems to produce a 404 error, I will look over it

**Reply by Zoey2936** [reply]
develop/beta/latest and develop-arm64 seem to work

**Reply by szaimen** [reply]
So only latest-arm64 is not working?

**Reply by Zoey2936** [reply]
latest-arm64 and beta-arm64

**Reply by Zoey2936** [reply]
they only have a `application/vnd.docker.distribution.manifest.list.v2+json`, no `application/vnd.docker.distribution.manifest.v2+json` and no `application/vnd.oci.image.index.v1+json`

**Reply by szaimen** [reply]
I see. So we need to change the deploy scripts to produce a application/vnd.docker.distribution.manifest.v2+json again...

**Reply by szaimen** [reply]
... For beta-arm64 and latest-arm64

**Reply by Zoey2936** [reply]
The problem occurs after copying the arm64 images, using: `docker buildx imagetools create --tag`, but the same line works for clamav (singel arch) and all multiarch images

**Reply by Zoey2936** [reply]
> I see. So we need to change the deploy scripts to produce a application/vnd.docker.distribution.manifest.v2+json again...

maybe use `docker pull`, `docker tag`, `docker push` again for them?

**Reply by Zoey2936** [reply]
or `docker manifest create nextcloud/$AIO_NAME\:beta-arm64 -a nextcloud/$AIO_NAME\:develop-arm64` and `docker manifest push nextcloud/$AIO_NAME\:beta-arm64`

**Reply by szaimen** [reply]
> or `docker manifest create nextcloud/$AIO_NAME\:beta-arm64 -a nextcloud/$AIO_NAME\:develop-arm64` and `docker manifest push nextcloud/$AIO_NAME\:beta-arm64`

Would appreciate this if it creates application/vnd.docker.distribution.manifest.v2+json

**Reply by Zoey2936** [reply]
https://github.com/nextcloud-releases/all-in-one/pull/34 I think it should work

**Reply by Zoey2936** [reply]
Btw. when https://github.com/docker/cli/pull/3994 is merged we could remove this line: https://github.com/nextcloud-releases/all-in-one/blob/eef7898f5c733b849ffe9ed2e88016f3afda9e9a/.github/workflows/build_images.yml#L71, but I think that would remove `application/vnd.docker.distribution.manifest.v2+json` manifests entirely, if it does this not, we could remove this line after it is merged and release and inside the ubuntu-latest workflow iso, but if it breaks it, we should wait a long time until nearly every AIO instance supports `application/vnd.oci.image.index.v1+json` manifests
